src/extractor/guidance/sliproad_handler.cpp

```diff
@@ -29,273 +32,720 @@ SliproadHandler::SliproadHandler(const IntersectionGenerator &intersection_gener
-                     intersection_node_id != intersection_node_one)
-                return 1;
-            else
-                return 0;
-        }
-        else
-            return 0;
-    }();
+    // The intersection `c` has to be reachable
+    if (!next_road.entry_allowed)
+    {
+        return intersection;
+    }
 
-    if (obvious_turn_index == 0)
+    // The road leading to the intersection (bc) has to continue from our source
+    if (!roadContinues(source_edge_id, next_road.eid))
+    {
         return intersection;
+    }
 
-    const auto &next_road = intersection[obvious_turn_index];
+    // Link-check for (bc) and later on (cd) which both are getting shortcutted by Sliproad
+    const auto is_potential_link = [this, next_road](const ConnectedRoad &road) {
+        if (!road.entry_allowed)
+            return false;
 
-    const auto linkTest = [this, next_road](const ConnectedRoad &road) {
-        return !node_based_graph.GetEdgeData(road.eid).roundabout && road.entry_allowed &&
-               angularDeviation(road.angle, STRAIGHT_ANGLE) <= 2 * NARROW_TURN_ANGLE &&
-               !hasRoundaboutType(road.instruction) &&
-               angularDeviation(next_road.angle, road.angle) >
-                   std::numeric_limits<double>::epsilon();
-    };
+        // Narrow turn angle for road (bd) and guard against data issues (overlapping roads)
+        auto is_narrow = angularDeviation(road.angle, STRAIGHT_ANGLE) <= 2 * NARROW_TURN_ANGLE;
+        auto same_angle = angularDeviation(next_road.angle, road.angle) //
+                          <= std::numeric_limits<double>::epsilon();
 
-    bool hasNarrow =
-        std::find_if(intersection.begin(), intersection.end(), linkTest) != intersection.end();
-    if (!hasNarrow)
-        return intersection;
+        if (!is_narrow || same_angle)
+            return false;
+
+        // Prevent from starting in or going onto a roundabout
+        auto onto_roundabout = hasRoundaboutType(road.instruction);
 
-    const auto source_edge_data = node_based_graph.GetEdgeData(source_edge_id);
-    // check whether the continue road is valid
-    const auto check_valid = [this, source_edge_data](const ConnectedRoad &road) {
-        const auto road_edge_data = node_based_graph.GetEdgeData(road.eid);
-        // Test to see if the source edge and the one we're looking at are the same road
+        if (onto_roundabout)
+            return false;
 
-        const auto same_name = !util::guidance::requiresNameAnnounced(
-            road_edge_data.name_id, source_edge_data.name_id, name_table, street_name_suffix_table);
+        const auto road_data = node_based_graph.GetEdgeData(road.eid);
 
-        return road_edge_data.road_classification == source_edge_data.road_classification &&
-               road_edge_data.name_id != EMPTY_NAMEID && source_edge_data.name_id != EMPTY_NAMEID &&
-               same_name && road.entry_allowed;
+        auto is_roundabout = road_data.roundabout;
```

Comment:
duplicated check? Within roundabouts, you should have only size two connections.
Outside of roundabouts/on exits you would see a roundabout type instruction.

Also, If it should be necessary, I'd prefer doing a single function style check for `isRoundabout` instead of two separate checks.